packages/bigtable/src/table.js

```diff
@@ -735,20 +735,51 @@ Table.prototype.getRows = function(options, callback) {
  * table.insert(entries, callback);
  *
  * //-
- * // If you don't provide a callback, an EventEmitter is returned. Listen for
- * // the error event to catch API and insert errors, and complete for when
- * // the API request has completed.
+ * // If the callback is omitted, we'll return a Promise.
  * //-
- * table.insert(entries)
+ * table.insert(entries).then(function() {
+ *   // All requested inserts have been processed.
+ * });
+ */
+Table.prototype.insert = function(entries, callback) {
+  entries = arrify(entries).map(propAssign('method', Mutation.methods.INSERT));
+
+  return this.mutate(entries, callback);
+};
+
+/**
+ * Insert or update rows in your table. It should be noted that gRPC only allows
+ * you to send payloads that are less than or equal to 4MB. If you're inserting
+ * more than that you may need to send smaller individual requests.
+ *
+ * Use this method over {bigtable/table#insert} if you want to handle insert
+ * errors as a readable object stream.
+ *
+ * @param {object|object[]} entries - List of entries to be inserted.
+ *     See {module:bigtable/table#mutate}.
+ *
+ * @example
+ * var entries = [
+ *  {
+ *     key: 'alincoln',
+ *     data: {
+ *       follows: {
+ *         gwashington: 1
+ *       }
+ *     }
+ *   }
+ * ];
+ *
+ * table.insertStream(entries)
  *   .on('error', console.error)
- *   .on('complete', function() {
+ *   .on('finish', function() {
```

Comment:
I'm find with renaming it something else, but obviously we need to separate this functionality some how. This is probably going to require new since this is pretty unique to our API. Any suggestions?